intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/normalizedcache/NormalizedCache.kt

```diff
@@ -31,11 +31,52 @@ data class NormalizedCache(
   )
 
   companion object {
+    /**
+     * Compares records by key, with `QUERY_ROOT` first, then by alphabetical order, taking numbers into account.
+     * E.g. `person.1` < `person.2` < `person.10` < `person.10.friend.1` < `person.10.friend.2`
+     */
     val RecordKeyComparator = Comparator<Record> { o1, o2 ->
+      val key1 = o1.key
+      val key2 = o2.key
       when {
-        o1.key == "QUERY_ROOT" -> -1
-        o2.key == "QUERY_ROOT" -> 1
-        else -> o1.key.compareTo(o2.key, ignoreCase = true)
+        key1 == "QUERY_ROOT" -> if (key2 == "QUERY_ROOT") 0 else -1
+        key2 == "QUERY_ROOT" -> 1
+        else -> {
+          var idx1 = 0
+          var idx2 = 0
+          while (true) {
+            val c1 = key1[idx1]
+            val c2 = key2[idx2]
+            if (c1.isDigit() && c2.isDigit()) {
+              var number1 = 0
+              var number2 = 0
```

Comment:
Make them BigInteger so that we're overflow proof. Not a huge deal but I think there's a law saying that if something bad _may_ happen, that something _will_ happen? 🤷

Reply:
Oh well nevermind, it's only list indices (not ids or other ints) so yea, if we have more than INT_MAX list items, we'll probably have more pressing matters 👍